### PR TITLE
Small fixes to make it work with Python 3

### PR DESCRIPTION
**Pull request by kartikmohta**
Note that two tests fail due to internal changes in Python 3:
- test_iterable_literals_eval: The order of keys in dict changed in Py3
- test_no_evaluation: ast.literal_eval('5 -2') gives 3 in Py3

### CODE REVIEW

**Comment by rhaschke** [comment]
~~I'm wondering whether these changes are really neccessary. Did you have specific issues running xacro with python3? Running the unittests on my side, I don't see any issues.~~
Sorry, the unittests didn't yet used python3 on my side. Now, I'm getting some errors too ;-)

**Comment by kartikmohta** [comment]
`cStringIO` and `dict.iteritems` are not present in Python 3, so these changes are required.

File: src/xacro/__init__.py [hunk 1]
```diff
@@ -950,7 +950,7 @@ def process_doc(doc,
     if do_check_order and symbols.redefined:
         warning("Document is incompatible to --inorder processing.")
         warning("The following properties were redefined after usage:")
-        for k, v in symbols.redefined.iteritems():
+        for k, v in symbols.redefined.items():
```
Comment (rhaschke) [inline src/xacro/__init__.py:949]: Although iteritems() is more efficient than items() in python2, I think for the sake of simplicity, we can switch to items().

File: xacro.py [hunk 2]
```diff
@@ -50,7 +50,7 @@
 this_dir_cwd = os.getcwd()
 os.chdir(cur_dir)
 # Remove this dir from path
-sys.path = filter(lambda a: a not in [this_dir, this_dir_cwd], sys.path)
+sys.path = [a for a in sys.path if a not in [this_dir, this_dir_cwd]]
```
Comment (rhaschke) [inline xacro.py:53]: This is definitely more pythonic ;-)

**Review by rhaschke — CHANGES_REQUESTED** [review]
Additionally to your requested changes, I had to
import xml.dom.minidom
in xmlutils.py. Strange, that you didn't notice that issue.

File: src/xacro/xmlutils.py [hunk 3]
```diff
@@ -31,6 +31,7 @@
 # Maintainer: Morgan Quigley <morgan@osrfoundation.org>
 
 import xml
```
Comment (rhaschke) [inline src/xacro/xmlutils.py]: The "import xml" can be removed then.

**Comment by rhaschke** [comment]
As soon as #161 is merged, I will add some more commits, cleanup and re-issue as a new PR.
I fixed everything for python3 already, such that the unittests pass:
https://travis-ci.org/ubi-agni/xacro

**Pull request closed without merging**